### PR TITLE
refactor(parser): swap arrow in favour of datetime in JP and occtonet

### PR DESCRIPTION
**Pull request by consideRatio**
## Issue

Resolves subtasks of #6135, for occtonet and JP, which had to be done at the same time to avoid datatype mixup.

I've tested this together with #7620, and the `fetch_price` parser seems to work well with this refactoring as well with that PR applied.

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
  ```
  poetry run test_parser "JP-HKD->JP-TH" && echo OK 1
  poetry run test_parser "JP-TH->JP-TK" && echo OK 2
  poetry run test_parser "JP-CB->JP-TK" && echo OK 3
  poetry run test_parser "JP-CB->JP-KN" && echo OK 4
  poetry run test_parser "JP-CB->JP-HR" && echo OK 5
  poetry run test_parser "JP-HR->JP-KN" && echo OK 6
  poetry run test_parser "JP-CG->JP-KN" && echo OK 7
  poetry run test_parser "JP-KN->JP-SK" && echo OK 8
  poetry run test_parser "JP-CG->JP-SK" && echo OK 9
  poetry run test_parser "JP-CG->JP-KY" && echo OK 10
  
  poetry run test_parser "JP-CB" && echo 1
  poetry run test_parser "JP-CG" && echo 2
  poetry run test_parser "JP-HKD" && echo 3
  poetry run test_parser "JP-HR" && echo 4
  poetry run test_parser "JP-KN" && echo 5
  poetry run test_parser "JP-KY" && echo 6
  poetry run test_parser "JP-ON" && echo 7
  poetry run test_parser "JP-SK" && echo 8
  poetry run test_parser "JP-TH" && echo 9
  poetry run test_parser "JP-TK" && echo 10
  
  poetry run test_parser "JP-CB" price && echo 1
  poetry run test_parser "JP-CG" price && echo 2
  poetry run test_parser "JP-HKD" price && echo 3
  poetry run test_parser "JP-HR" price && echo 4
  poetry run test_parser "JP-KN" price && echo 5
  poetry run test_parser "JP-KY" price && echo 6
  poetry run test_parser "JP-ON" price && echo 7
  poetry run test_parser "JP-SK" price && echo 8
  poetry run test_parser "JP-TH" price && echo 9
  poetry run test_parser "JP-TK" price && echo 10
  ```
- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: parsers/occtonet.py [hunk 1]
```diff
@@ -110,10 +117,15 @@ def fetch_exchange_forecast(
...
+    if target_datetime is None:
+        target_datetime = now
+    else:
+        target_datetime = target_datetime.astimezone(ZONE_INFO)
 
-    query_datetime = arrow.get(target_datetime).to("Asia/Tokyo").strftime("%Y/%m/%d")
+    query_datetime = target_datetime.strftime("%Y/%m/%d")
 
-    if query_datetime > arrow.get().to("Asia/Tokyo").strftime("%Y/%m/%d"):
+    if query_datetime > now.strftime("%Y/%m/%d"):
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply parsers/occtonet.py:128]: This is done in many other places, and I think it should be fine with this formatting.

[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply parsers/occtonet.py:128]: It should be fine for now but in general it is best to do a native comparison, I don't know why this was done the way it is. This string conversion and comparison should also be less performant, which is not really an issue but always nice to fix in the future.

File: parsers/occtonet.py [hunk 2]
```diff
@@ -207,7 +219,17 @@ def get_form_data(session: Session, exchange_id: int, datetime: str) -> dict[str
...
+        Expects arguments like "2024/12/23 00:05", including the edge case 24:00
+        which is the same as 00:00 of the subsequent date.
+        """
+        dt = None
+        if "24:00" in str_dt:
+            str_dt = str_dt.replace("24:00", "00:00")
+            dt = datetime.strptime(str_dt, "%Y/%m/%d %H:%M") + timedelta(days=1)
+        else:
+            dt = datetime.strptime(str_dt, "%Y/%m/%d %H:%M")
+        return dt.replace(tzinfo=ZONE_INFO)
```
Comment (VIKTORVAV99) [inline parsers/occtonet.py:232]: Nice catch!

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

There seems to be some existing type issues but we can fix those in a later PR. Nice work!

**Pull request merged**